### PR TITLE
Update codeowners to be consistent with specs and oteps

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -12,4 +12,14 @@
 #  https://help.github.com/en/articles/about-code-owners
 #
 
-*  @bogdandrutu @carlosalberto @SergeyKanzhelev @yurishkuro @tedsuo @iredelmeier @arminru @c24t @reyang @tigrannajaryan @jmacd
+# Global owners, will be the owners for everything in the repo. Membership is tracked via https://github.com/open-telemetry/community/blob/master/community-members.md 
+*   @open-telemetry/specs-approvers
+
+# Trace owners (global + trace)
+opentelemetry/proto/trace/    @open-telemetry/specs-approvers @open-telemetry/specs-trace-approvers
```
Comment (tigrannajaryan) [inline CODEOWNERS:19]: Can you also add `opentelemetry/proto/collector/trace/v1/` and similarly for other signals?

Comment (bogdandrutu) [reply CODEOWNERS:19]: I feel there are very few changes there, they are mostly constant that files.

Comment (tigrannajaryan) [reply CODEOWNERS:19]: OK, we can add later if needed.

**Comment by Oberon00** [comment]
I am not sure if the collector/trace/ directories, etc. should also be added, but with the current content, it seems OK not to have the area-specific code owners applied there (since it would more likely belong to a "collector" group than a "trace" group for example).

**Pull request merged**